lighthouse-cli/test/smokehouse/smokehouse.js

```diff
@@ -111,6 +111,10 @@ function findDifference(path, actual, expected) {
     return null;
   }
 
+  if (expected instanceof RegExp && expected.test(actual)) {
```

Comment:
`expected.test()` will do `actual.toString()` internally, so this check should be good for any value of `actual`, but if it fails this check, `typeof expected` will still be `'object'`, so it could pass the conditional below and then try to recurse on the regexp object instead of returning the difference. This could lead to a false positive result since `Object.keys(/ul.versionlist/)` is `[]`, so any value of `actual` that has `typeof` `'object'` will pass. e.g. `findDifference('path', {}, /whatever/)` will find no difference.

Simplest thing would be to also add `|| expected instanceof RegExp` to the conditional below.